kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -503,6 +533,22 @@ namespace Kratos
         }
     }
 
+    template<std::size_t TDim>
+    void CalculateDiscontinuousDistanceToSkinProcess<TDim>::ReplaceZeroDistances(
+        Vector& rElementalDistances)
+    {
+        const bool use_negative_epsilon = mOptions.Is(CalculateDiscontinuousDistanceToSkinProcessFlags::USE_NEGATIVE_EPSILON_FOR_ZERO_VALUES);
+        const double multiplier = (use_negative_epsilon)  ? -mZeroToleranceMultiplier : mZeroToleranceMultiplier;
+        for (auto& r_distance : rElementalDistances) {
```

Comment:
```suggestion
        for (double& r_distance : rElementalDistances) {
```
Avoid auto for simple types.